src/coins.cpp

```diff
@@ -197,23 +216,31 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
             }
         }
     }
+
+    // MWEB: Flushes mw coins
+    derivedView->Flush(nullptr);
+
     hashBlock = hashBlockIn;
     return true;
 }
 
 bool CCoinsViewCache::Flush() {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock);
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, mweb_view);
     cacheCoins.clear();
     cachedCoinsUsage = 0;
     return fOk;
 }
 
-void CCoinsViewCache::Uncache(const COutPoint& hash)
+void CCoinsViewCache::Uncache(const OutputIndex& coin)
 {
-    CCoinsMap::iterator it = cacheCoins.find(hash);
-    if (it != cacheCoins.end() && it->second.flags == 0) {
-        cachedCoinsUsage -= it->second.coin.DynamicMemoryUsage();
-        cacheCoins.erase(it);
+    if (coin.which() == 0) {
```

Comment:
Is is safer to use .type() == typeid ? In case the types are reordered.

Reply:
Types should never be reordered, but I do find typeid to be more readable. I'll change this.